plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMergeSink.java

```diff
@@ -324,7 +325,9 @@ private ReaderPageSource createParquetPageSource(Path path)
 
         private FileDeletion(List<String> partitionValues)
         {
-            this.partitionValues = ImmutableList.copyOf(requireNonNull(partitionValues, "partitionValues is null"));
+            // This can't be an ImmutableList, because Delta Lake allows NULL partition values.
```

Comment:
I think we can simplify this to
```java
// use ArrayList since Delta Lake allows NULL partition values
```
This provides context as to why we can't use `ImmutableList` or `List.copyOf()`, without needing to say so explicitly.

Reply:
Changed to:

```
            // Use ArrayList since Delta Lake allows NULL partition values, and wrap it in
            // an unmodifiableList.
```